kep/640-error-pods-rescheduling/README.md

```diff
@@ -0,0 +1,166 @@
+For instance, a pod might get OOM killed due to insufficient host-node memory, or it could be rejected by kubelet during startup because of mismatches in NUMA or other device requirements. 
+In such cases, local restarts are ineffective. Manual intervention becomes necessary, leading to increased operational complexity.
+
+This situation contradicts Kubernetes' claim of automatic fault tolerance capabilities, diminishing the reliability and high availability of the service. 
+Therefore, we are motivated to develop rescheduling plugins based on the proposed scheduler framework. These plugins can reschedule pods that encounter errors to different nodes, excluding the nodes where they were previously scheduled.
+
+### Goals
+
+* Reschedule already scheduled pods (encounter some errors) to other nodes automatically.
+* Implement pod rescheduling plugins in PreFilter extension point to prefilter nodes that the pod has already scheduled.
+* Pod rescheduling plugin in this proposal has no conflict with default scheduler's plugins.
+
+### Non-Goals
+
+* Reschedule pods controlled by daemonset or other workload resources which don't exist in default k8s controller.
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+A pod submitted to the Kubernetes cluster initially operates smoothly, but over time, as the memory on the node hosting the pod becomes insufficient and the pod's memory requirements grow, it eventually gets CrashLoopBackOff due to out-of-memory (OOM) issues. 
+Consequently, we have to manually relocate the pod to other nodes. But this pod may schedule to the same node with the default scheduler.
+
+#### Story 2
+
+After a pod is submitted to the Kubernetes cluster and successfully scheduled, it may encounter startup issues due to mismatches in hardware resources on the node, 
+such as NUMA or device-related information. In such cases, we need to manually reschedule the pod to a different node. But this pod may schedule to the same node with the default scheduler.
+
+## Design Details
+
+Our scheme is to design two components as outlined in the diagram and described below: Rescheduler-controller and Podrescheduling. 
+Users can configure the `scheduling-retries` (maximum rescheduling retry count) in the metadata annotations of workload resources, such as Deployment and StatefulSet. 
+Rescheduler-controller is responsible for monitoring pods with abnormal states in the cluster. For each abnormal pod, 
+the rescheduler-controller checks the current scheduling count (defined as `currentReschedulingTimes`) recorded in the metadata annotations of its associated workload resource. 
+If the current scheduling count does not exceed the user-configured `scheduling-retries` (maximum rescheduling retry count), 
+it increments the current scheduling count by 1 and records the current node where the pod is located in the scheduled nodes list (eg: defined as `deployScheduledHosts`). 
```

Comment:
logic seems fine, but for the sake of clarity I'd mention earlier and more explicitly that the most important extra annotations are both `currentReschedulingTimes` and `deployScheduledHosts`. The flow of the tests initially highlights the former and only later introduces the latter on the go.
Additionally I'd use a consistent style between `camelCase` and `dash-separate-words`.

Reply:
Ok, i will use a consistent style between `camelCase` and `dash-separate-words`.